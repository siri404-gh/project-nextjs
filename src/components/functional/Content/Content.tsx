import React, { FunctionComponent } from 'react';
import cn from 'classnames';
import {
  Paper,
  makeStyles,
} from '@material-ui/core';

interface Props {
  className?: string,
  children?: JSX.Element,
}

const useStyles = makeStyles(theme => ({
  content: {
    padding: 0,
    marginBottom: 0,
  },
  paper: {
    padding: 15,
    [theme.breakpoints.up('sm')]: {
      padding: 20,
    },
    [theme.breakpoints.down('xs')]: {
      borderRadius: 0,
    },
  },
}));

const Content: FunctionComponent<Props> = ({ className, children }) => {
  const classes = useStyles();

  return (
    <main className={cn(classes.content, className)}>
      <Paper className={classes.paper}>
        {children}
      </Paper>
    </main>
  );
};

Content.defaultProps = {
  className: '',
  children: null,
};

export default Content;
