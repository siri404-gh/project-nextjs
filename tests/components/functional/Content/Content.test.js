import { render, unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';

import Content from '@/components/functional/Content/Content';

describe('Content', () => {
  let container;

  beforeEach(() => {
    container = document.createElement('section');
    document.body.appendChild(container);
  });

  afterEach(() => {
    unmountComponentAtNode(container);
    container.remove();
    container = null;
  });

  it('Should render without props', () => {
    act(() => {
      render(<Content />, container);
    });
    expect(container).toMatchSnapshot();
  });
});
