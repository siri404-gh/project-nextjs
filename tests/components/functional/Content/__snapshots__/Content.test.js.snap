// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content Should render without props 1`] = `
<section>
  <main
    class="makeStyles-content-1"
  >
    <div
      class="MuiPaper-root makeStyles-paper-2 MuiPaper-elevation1 MuiPaper-rounded"
    />
  </main>
</section>
`;
